build.yaml

```diff
@@ -3147,6 +3185,57 @@ steps:
       - make_docs
     clouds:
       - gcp
+  - kind: runImage
+    name: mirror_hailgenetics_images
+    image: quay.io/skopeo/stable:v1.11.1
+    script: |
+      set -ex
+
+      REGISTRY={{ global.docker_prefix.split('/')[0] }}
+
+      set +x
+      {% if global.cloud == "gcp" %}
+      cat /registry-push-credentials/credentials.json | base64 -w 0 | skopeo login -u _json_key_base64 --password-stdin $REGISTRY
+      {% elif global.cloud == "azure" %}
+      dnf install -y jq
+      USERNAME=$(cat /registry-push-credentials/credentials.json | jq -jr '.appId')
+      cat /registry-push-credentials/credentials.json | jq -jr '.password' | skopeo login -u $USERNAME --password-stdin $REGISTRY
+      {% else %}
+      echo "unknown cloud {{ global.cloud }}"
+      exit 1
+      {% endif %}
+      set -x
+
+      copy_if_not_present() {
+        src_image=$1
+        dest_image=$2
+        if skopeo inspect "docker://$2";
+        then
+          echo "$2 already exists, doing nothing"
+        else
+          echo "$2 does not exist, copying $1 to $2"
+          skopeo copy "docker://$1" "docker://$2"
+        fi
+      }
+
+      copy_if_not_present {{ hailgenetics_hail_image.image}} {{ global.docker_prefix }}/hailgenetics/hail:$(cat /io/hail_pip_version)
+      copy_if_not_present {{ hailgenetics_hailtop_image.image}} {{ global.docker_prefix }}/hailgenetics/hailtop:$(cat /io/hail_pip_version)
```

Comment:
So, this is decoupled from the `deploy` step. We can fail to release 0.2.X but still push hailgenetics/hail:0.2.X. That docker image will have a different SHA from the PyPI release. I don't love that.

Moreover, if, say, AUS runs a deploy on a SHA that's not a tagged SHA, they'll get a hailgenetics/hail image with a tag like 0.2.X but whose SHA is not the SHA tagged with 0.2.X. I dislike that.

What do you think of cloning the images tagged with `/io/hail_pip_version` from docker hub, if it exists?

Reply:
I would much prefer that. I think I went this approach because I was worried about say Azure racing GCP and getting to this step before GCP publishes the image to dockerhub. But ensuring that the same SHAs are in use everywhere seems worth Azure possibly not having the newest dockerhub image for a single deployment. If this is ever an actual issue we could just run a cronjob or webhook or something.

Reply:
Hm, actually this would apply to GCP too, as we don't want to necessarily condition it on running after the `deploy` step (AUS folks don't run that nor should they).

Reply:
@danking Wondering if the fact that this will also affect our GCP instance changes your perspective.

Reply:
Generally, I'd rather get it late than get it wrong.

That said, the `deploy` step already copies into us-docker.pkg.dev/hail-vdc, so I don't think this affects GCP.

Reply:
Ah, right, great. I forgot we were keeping it in the `deploy` step. Fine with this then (and agreed on better late than wrong)